File: libs/validators/python/validators.py
"""
Shared validators — extracted from:
  - apps/user-service/app/services/user_service.py
  - apps/notification-service/internal/service/email_utils.go (Go equivalent)

Lab 3 extraction target. Import this in user-service to replace the local copy.
"""
import re
from datetime import datetime


def validate_email(email: str) -> bool:
    """Validate email address format. RFC 5322 simplified."""
    pattern = r'^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$'
    return bool(re.match(pattern, email))


def format_date(dt: datetime, fmt: str = "%Y-%m-%dT%H:%M:%SZ") -> str:
    """Format a datetime to ISO 8601 string."""
    return dt.strftime(fmt)

File: libs/validators/go/validators.go
// Package validators provides shared validation utilities extracted from:
//   - apps/notification-service/internal/service/email_utils.go
//   - apps/user-service/app/services/user_service.py (Python equivalent)
//
// Lab 3 extraction target. Import this in notification-service to replace the local copy.
package validators

import (
	"regexp"
	"time"
)

var emailRegex = regexp.MustCompile(`^[a-zA-Z0-9._%+\-]+@[a-zA-Z0-9.\-]+\.[a-zA-Z]{2,}$`)

// ValidateEmail returns true if the email address matches RFC 5322 simplified pattern.
func ValidateEmail(email string) bool {
	return emailRegex.MatchString(email)
}

// FormatDate formats a time.Time to ISO 8601 UTC string.
func FormatDate(t time.Time) string {
	return t.UTC().Format(time.RFC3339)
}
